src/emulation/sync.ts

```diff
@@ -463,17 +469,36 @@ export function readdirSync(
 		}
 		entries.push(entry);
 	}
-	return entries.map((entry: string) => {
-		if (typeof options == 'object' && options?.withFileTypes) {
-			return new Dirent(entry, statSync(join(path.toString(), entry)));
-		}
 
-		if (options == 'buffer' || (typeof options == 'object' && options?.encoding == 'buffer')) {
-			return Buffer.from(entry);
+	// Iterate over entries and handle recursive case if needed
+	const values: (string | Dirent | Buffer)[] = [];
+	for (const entry of entries) {
+		const fullPath = join(path, entry);
+		const entryStat = statSync(fullPath);
+
+		if (options?.withFileTypes) {
```

Comment:
You can convert this to a ternary, which is a little more concise:

```ts
values.push(options?.withFileTypes ? new Dirent(entry, entryStat) : options?.encoding == 'buffer' ? Buffer.from(entry) : entry);
```

It is slightly less readable, however.  You can choose.

Whichever you choose, please use the same format (a ternary or if statements) in both the sync and async varients for consistency.

Reply:
Sounds good.